src/posix/platform/spi_interface.cpp

```diff
@@ -91,32 +91,91 @@ SpiInterface::SpiInterface(SpinelInterface::ReceiveFrameCallback aCallback,
 {
 }
 
-otError SpiInterface::Init(const otPlatformConfig &aPlatformConfig)
+otError SpiInterface::Init(Arguments &aArguments)
 {
-    VerifyOrDie(aPlatformConfig.mSpiAlignAllowance <= kSpiAlignAllowanceMax, OT_EXIT_FAILURE);
+    const char *spiGpioIntDevice;
+    const char *spiGpioResetDevice;
+    uint8_t     spiGpioIntLine     = 0;
+    uint8_t     spiGpioResetLine   = 0;
+    uint8_t     spiMode            = OT_PLATFORM_CONFIG_SPI_DEFAULT_MODE;
+    uint32_t    spiSpeed           = SPI_IOC_WR_MAX_SPEED_HZ;
+    uint32_t    spiResetDelay      = OT_PLATFORM_CONFIG_SPI_DEFAULT_RESET_DELAY_MS;
+    uint16_t    spiCsDelay         = OT_PLATFORM_CONFIG_SPI_DEFAULT_CS_DELAY_US;
+    uint8_t     spiAlignAllowance  = OT_PLATFORM_CONFIG_SPI_DEFAULT_ALIGN_ALLOWANCE;
+    uint8_t     spiSmallPacketSize = OT_PLATFORM_CONFIG_SPI_DEFAULT_SMALL_PACKET_SIZE;
 
-    mSpiCsDelayUs       = aPlatformConfig.mSpiCsDelay;
-    mSpiSmallPacketSize = aPlatformConfig.mSpiSmallPacketSize;
-    mSpiAlignAllowance  = aPlatformConfig.mSpiAlignAllowance;
+    spiGpioIntDevice   = aArguments.GetValue("gpio-int-device");
+    spiGpioResetDevice = aArguments.GetValue("gpio-reset-device");
+    if (!spiGpioIntDevice || !spiGpioResetDevice)
+    {
+        DieNow(OT_EXIT_INVALID_ARGUMENTS);
+    }
+
+    if (aArguments.GetValue("gpio-int-line"))
+    {
+        spiGpioIntLine = static_cast<uint8_t>(atoi(aArguments.GetValue("gpio-int-line")));
+    }
+    else
+    {
+        DieNow(OT_EXIT_INVALID_ARGUMENTS);
+    }
+    if (aArguments.GetValue("gpio-reset-line"))
+    {
+        spiGpioResetLine = static_cast<uint8_t>(atoi(aArguments.GetValue("gpio-reset-line")));
+    }
+    else
+    {
+        DieNow(OT_EXIT_INVALID_ARGUMENTS);
+    }
+    if (aArguments.GetValue("spi-mode"))
+    {
+        spiMode = static_cast<uint8_t>(atoi(aArguments.GetValue("spi-mode")));
+    }
+    if (aArguments.GetValue("spi-speed"))
+    {
+        spiSpeed = static_cast<uint32_t>(atoi(aArguments.GetValue("spi-speed")));
+    }
+    if (aArguments.GetValue("spi-reset-delay"))
+    {
+        spiResetDelay = static_cast<uint32_t>(atoi(aArguments.GetValue("spi-reset-delay")));
+    }
+    if (aArguments.GetValue("spi-cs-delay"))
+    {
+        spiCsDelay = static_cast<uint16_t>(atoi(aArguments.GetValue("spi-cs-delay")));
+    }
+    if (aArguments.GetValue("spi-align-allowance"))
+    {
+        spiAlignAllowance = static_cast<uint8_t>(atoi(aArguments.GetValue("spi-align-allowance")));
+    }
+    if (aArguments.GetValue("spi-small-packet"))
+    {
+        spiSmallPacketSize = static_cast<uint8_t>(atoi(aArguments.GetValue("spi-small-packet")));
```

Comment:
Another thing that `getopt_long()` would help is to detect missing required value for an option when the `option` struct specified the `required_argument` for an option entry. I think in this case `getopt_long()` returns `?` which we handle as error in the code and exit.

I think with the `GetValue()` we cannot detech that and fallback to empty string as value if none is given.

Reply:
The current behavior is that we return `nullptr` for keys not specified in the url. The consumer of the arguments takes the responsibility of ensuring the required ones. It will be a fancy feature to report un-recognized arguments but maybe we can leave for future improvements since ot-br-posix is pending on this feature to fully support all possible radio interfaces.